### PR TITLE
[MI-2581] Fix issues given by QA

### PR DESCRIPTION
**Pull request by ayusht2810**
image
image
image
image
image
image
image

Issues fixed:

1. Add channel validations in the incident modal and update the incident modal API
2. Update default values of impact and urgency in the incident modal
3. Fixing text area style on different themes
4. Fixing the loader issue in the Catalog Request modal
5. Fix closing of the modal after clicking on Create Request modal button
6. Fixing post menu theme options for opening incident modal and removing pre-filling of short-description
7. Fixing comments in the comment modal

### CODE REVIEW

File: server/plugin/client.go [hunk 1]
```diff
@@ -271,9 +271,13 @@ func (c *client) GetMe(userEmail string) (*serializer.ServiceNowUser, int, error
 }
 
 func (c *client) CreateIncident(incident *serializer.IncidentPayload) (*serializer.IncidentResponse, int, error) {
+	queryParams := url.Values{
+		constants.SysQueryParamDisplayValue: {"true"},
```
Comment (Kshitij-Katiyar) [inline server/plugin/client.go:275]: is this gonna be true every time ?

**Review by Kshitij-Katiyar — APPROVED** [review]

File: server/plugin/client.go (hunk 1 above)
Comment (ayusht2810) [reply server/plugin/client.go:275]: yes

File: webapp/src/containers/createIncident/index.tsx [hunk 2]
```diff
@@ -218,6 +231,12 @@ const CreateIncident = () => {
         }
     }, [open, refetchIncidentFields]);
 
+    useEffect(() => {
+        if (channel) {
```
Comment (manojmalik20) [inline webapp/src/containers/createIncident/index.tsx:235]: Isn't their any other `useEffect` in which you can add this logic?

**Review by manojmalik20 — APPROVED** [review]

File: webapp/src/containers/createIncident/index.tsx (hunk 2 above)
Comment (ayusht2810) [reply webapp/src/containers/createIncident/index.tsx:235]: no, this is the second useEffect used in this component

**Review by raghavaggarwal2308 — APPROVED** [review]

File: server/plugin/api_test.go [hunk 3]
```diff
@@ -1906,6 +1906,7 @@ func TestAPICreateIncident(t *testing.T) {
 		RequestBody          string
 		SetupAPI             func(*plugintest.API)
 		SetupClient          func(client *mock_plugin.Client)
+		SetupPlugin          func()
```
Comment (manojmalik20) [inline server/plugin/api_test.go:1909]:
```suggestion
		SetupPlugin          func(p *Plugin)
```

Comment (ayusht2810) [reply server/plugin/api_test.go:1909]: we are also mocking some functions of different type. so I think it's okay to keep it like this

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**